### ISSUE TITLE
[no-input-rename] reports false positive when using required inputs.  

### ISSUE THREAD

**Issue by JeanMeche** [angular-eslint/angular-eslint#1355]
**Description and reproduction of the issue**

Angular v16 (currently in RC), provides a new API to specify required `@Input()`.

```TS
@Input({ required: true }) name: string;
```

The rule `"@angular-eslint/no-input-rename": "error",` on the latest 15 version of `angular-eslint`

Throws 

`Input bindings should not be aliased (https://angular.io/guide/styleguide#style-05-13)  @angular-eslint/no-input-rename`

**Comment by DibyodyutiMondal** [comment]
Required inputs are likely going to be adopted fast. 

Here is another reproduction:
```typescript
import { Directive, HostBinding, Input } from '@angular/core';
import { FormControl } from '@angular/forms';

@Directive({
  standalone: true,
  selector: '[formFieldControl]',
})
export class FormFieldDirective {
  @Input({required: true}) formFieldControl?: FormControl;

  @HostBinding('class') private get class() {
    return {
      'form-field': true,
      'invalid': this.formFieldControl?.invalid ?? false,
      'dirty': this.formFieldControl?.dirty ?? false
    }
  }
}
```

![image](https://user-images.githubusercontent.com/50345677/233757921-414deda7-45b8-4120-81f0-b46d534d4359.png)

**Comment by DibyodyutiMondal** [comment]
I found a fix: change the INPUT_ALIAS selector by adding `parent.key.name='alias'` at the end.

```diff
- export const INPUT_ALIAS = `:matches(PropertyDefinition, MethodDefinition[kind='set']) ${INPUT_DECORATOR} ${LITERAL_OR_TEMPLATE_ELEMENT}`; 
+ export const INPUT_ALIAS = `:matches(PropertyDefinition, MethodDefinition[kind='set']) ${INPUT_DECORATOR} ${LITERAL_OR_TEMPLATE_ELEMENT}[parent.key.name='alias']; 
```

This is the line we should change:

https://github.com/angular-eslint/angular-eslint/blob/86ebc862237739cf44886b35f0f43235275fa8b6/packages/utils/src/eslint-plugin/selectors.ts#L55

**Comment by DibyodyutiMondal** [comment]
Those unwilling to wait for the fix can edit their node modules:
`node_modules/@angular-eslint/utils/dist/eslint-plugin/selectors.js`

**Comment by DibyodyutiMondal** [comment]
I found a fix: change the `INPUT_ALIAS` selector by adding the following at the end.
`:matches([parent.callee.name='Input'],[parent.key.name='alias'])` 
(without any preceding space before the ':matches')

Explanation: 
1. `parent.callee.name` will catch decorators which use a string config `@Input('myAlias')`
2. `parent.key.name` will catch decorators which use object config `@Input({ alias: 'myAlias'}), but will only work on the alias property of the config

```diff
- export const INPUT_ALIAS = `:matches(PropertyDefinition, MethodDefinition[kind='set']) ${INPUT_DECORATOR} ${LITERAL_OR_TEMPLATE_ELEMENT}`; 
+ export const INPUT_ALIAS = `:matches(PropertyDefinition, MethodDefinition[kind='set']) ${INPUT_DECORATOR} ${LITERAL_OR_TEMPLATE_ELEMENT}:matches([parent.callee.name='Input'],[parent.key.name='alias'])`; 
```

This is the line we should change:

https://github.com/angular-eslint/angular-eslint/blob/86ebc862237739cf44886b35f0f43235275fa8b6/packages/utils/src/eslint-plugin/selectors.ts#L55


Those unwilling to wait for the fix can change their node_modules:
`node_modules/@angular-eslint/utils/dist/eslint-plugin/selectors.js`


#### New bahaviour after the change:

![image](https://user-images.githubusercontent.com/50345677/233760745-5aa47d4e-1337-411c-99c4-709cb24a1441.png)

**Comment by JamesHenry** [comment]
v16 is now the latest release and this issue has been addressed
